charms/kfp-persistence/src/components/pebble_components.py

```diff
@@ -0,0 +1,95 @@
+import dataclasses
+import logging
+from typing import Dict
+
+from charmed_kubeflow_chisme.components.pebble_component import PebbleServiceComponent
+from ops import ActiveStatus, StatusBase, WaitingStatus
+from ops.pebble import Layer
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class PesistenceAgentServiceConfig:
+    """Defines configuration for PersistenceAgent Service."""
+
+    KFP_API_SERVICE_NAME: str
+    NAMESPACE: str
+
+
+class PersistenceAgentPebbleService(PebbleServiceComponent):
+    """Pebble Service for Persistence Agent Container."""
+
+    def __init__(
+        self,
+        *args,
+        environment: Dict[str, str],
+        **kwargs,
+    ):
+        """Initialize component."""
+        super().__init__(*args, **kwargs)
+        self._environment = environment
+
+    def get_layer(self) -> Layer:
+        """Pebble configuration layer for persistenceagent.
+
+        This method is required for subclassing PebbleServiceComponent
+        """
+        logger.info("PersistenceAgentContainer: create layer")
+
+        # retrieve up-to-date service configuration as setup by charm
+        try:
+            service_config: PesistenceAgentServiceConfig = self._inputs_getter()
+        except Exception as err:
+            logger.error(f"PersistenceAgentContainer: configuration is not provided: {err}")
+            return None
+
+        if len(service_config.KFP_API_SERVICE_NAME) == 0 or len(service_config.NAMESPACE) == 0:
+            logger.info("PersistenceAgentContainer: configuration is not valid")
+            return None
```

Comment:
I'm not sure if these should return None, or raise.  If you haven't already, we might want to test this to see what's best